firewood/src/shale/compact.rs

```diff
@@ -24,7 +24,9 @@ pub struct CompactHeader {
 }
 
 impl CompactHeader {
-    pub const MSIZE: u64 = 17;
+    const IS_FREED_OFFSET: usize = 8;
```

Comment:
We can do sooo much better than this once offset_of is ready. It's slated for the next version of rust. On nightly, you can do:
```rust
    const IS_FREED_OFFSET: usize = core::mem::offset_of!(CompactHeader, is_freed);
```

I think it would be better to just wait for that to be on stable. It's "any day now".

Also, there is NO reason to have the is_freed bool before DiskAddress. This means unaligned accesses or padding, so swapping these two would be a good idea.

Since we are relying on the internal representation not being touched, we also need `#[repr(C)]` on `CompactHeader`. Otherwise, the compiler can (and in some cases will) reorder things or pack them differently on different architetures. Fortunately we're mostly on ix86 right now or this code may not even work.